ui/app/controllers/vault/cluster/access/methods.js

```diff
@@ -4,22 +4,72 @@
  */
 
 import Controller from '@ember/controller';
-import { task } from 'ember-concurrency';
+import { dropTask } from 'ember-concurrency';
 import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
 
-export default Controller.extend({
-  flashMessages: service(),
+export default class VaultClusterAccessMethodsController extends Controller {
+  @service flashMessages;
 
-  queryParams: {
-    page: 'page',
-    pageFilter: 'pageFilter',
-  },
+  @tracked authMethodOptions = [];
+  @tracked selectedAuthType = null;
+  @tracked selectedAuthName = null;
 
-  page: 1,
-  pageFilter: null,
-  filter: null,
+  queryParams = ['page, pageFilter'];
 
-  disableMethod: task(function* (method) {
+  page = 1;
+  pageFilter = null;
+  filter = null;
+
+  get authMethodList() {
+    // return an options list to filter by engine type, ex: 'kv'
+    if (this.selectedAuthType) {
+      // check first if the user has also filtered by name.
+      if (this.selectedAuthName) {
+        return this.model.filter((method) => this.selectedAuthName === method.id);
+      }
```

Comment:
I'm curious if we should be filtering by both type and name if they are both set? I think there could be a case where the same name is used for different types.

Reply:
For secret engines you can't have the same name as used by another secret engine. We do some validation work on that and say as much. And if they passed validation, then the API would return the same error.

But great question for auth methods. I had to check if this was the same. Turns out it is. We don't do any validation but we do return the API error.
![image](https://github.com/hashicorp/vault/assets/6618863/50daa1d9-8eaa-4763-bce3-c531e36ade4d)

So names are individualized across all auth methods. Which I believes answers your question?

Reply:
Ok good to know thanks! Maybe nice to add a comment in there stating as much so it's clear why type is ignored if name is set.